src/Sentry.OpenTelemetry/SentrySpanProcessor.cs

```diff
@@ -0,0 +1,311 @@
+    }
+
+    internal static Transaction TransactionProcessor(Transaction transaction)
+    {
+        var activity = Activity.Current;
+        if (activity != null)
+        {
+            var trace = transaction.Contexts.Trace;
+            trace.TraceId = activity.TraceId.AsSentryId();
+            trace.SpanId = activity.SpanId.AsSentrySpanId();
+        }
+
+        return transaction;
+    }
+
+    /// <inheritdoc />
+    public override void OnStart(Activity data)
+    {
+        if (data.ParentSpanId != default && _map.TryGetValue(data.ParentSpanId, out var parentSpan))
+        {
+            // We can find the parent span - start a child span.
+            var context = new SpanContext(
+                data.SpanId.AsSentrySpanId(),
+                data.ParentSpanId.AsSentrySpanId(),
+                data.TraceId.AsSentryId(),
+                data.OperationName,
+                data.DisplayName,
+                null,
+                null)
+            {
+                Instrumenter = Instrumenter.OpenTelemetry
+            };
+
+            var span = (SpanTracer)parentSpan.StartChild(context);
+            span.StartTimestamp = data.StartTimeUtc;
+            _map[data.SpanId] = span;
+        }
+        else
+        {
+            // If a parent exists at all, then copy its sampling decision.
+            bool? isSampled = data.HasRemoteParent ? data.Recorded : null;
+
+            // No parent span found - start a new transaction
+            var transactionContext = new TransactionContext(
+                data.SpanId.AsSentrySpanId(),
+                data.ParentSpanId.AsSentrySpanId(),
+                data.TraceId.AsSentryId(),
+                data.DisplayName,
+                data.OperationName,
+                data.DisplayName,
+                null,
+                isSampled,
+                isSampled)
+            {
+                Instrumenter = Instrumenter.OpenTelemetry
+            };
+
+            var baggageHeader = BaggageHeader.Create(data.Baggage.WithValues());
+            var dynamicSamplingContext = baggageHeader.CreateDynamicSamplingContext();
+            var transaction = (TransactionTracer)_hub.StartTransaction(
+                transactionContext, new Dictionary<string, object?>(), dynamicSamplingContext
+                );
+            transaction.StartTimestamp = data.StartTimeUtc;
+            _map[data.SpanId] = transaction;
+        }
+    }
+
+    /// <inheritdoc />
+    public override void OnEnd(Activity data)
+    {
+        // Make a dictionary of the attributes (aka "tags") for faster lookup when used throughout the processor.
+        var attributes = data.TagObjects.ToDictionary();
+
+        if (attributes.TryGetTypedValue("http.url", out string url) && _hub.IsSentryRequest(url))
+        {
+            // TODO: will this leave the span dangling?
+            _map.TryRemove(data.SpanId, out _);
```

Comment:
The OnEnd notification gets triggered by `Activity.Dispose()`. So in applications that aren't correctly disposing of their Activities, we could end up accumulating spans... in that case though, a `ConditionalWeakTable` probably wouldn't help us since the Activity wouldn't be garbage collected and the reference would remain active in the `ConditionalWeakTable`.

However, looking at the code above, I'm thinking we're probably leaving this check until too late. It's checking to see if the Activity relates to an outbound request to the Sentry DSN and, if so, it removes the span from the map without finishing it. It seems like it would make more sense never to create spans for these Activities  in the first place.

Reply:
Oh snap. Not being able to filter spans in `OnStart` sounds tricky? If a `SentrySpan` ends up as transaction and then gets discarded in `OnEnd`, doesn't that mess up all the child spans?

Reply:
> If a SentrySpan ends up as transaction and then gets discarded in OnEnd, doesn't that mess up all the child spans?

If that happened (and I'm not sure if it could) then none of the spans that were children of the SentrySpan that was a transaction would be sent to Sentry... since the Transaction would just be discarded and never finished. 

https://github.com/getsentry/sentry-dotnet/pull/2452 could have the side effect of sending these transactions even though we didn't intend them to be sent though... something to consider but we'd need to merge one or the other of these pull requests so that they were in the same codebase in order to resolve that. My inclination is to merge the OpenTelemetry stuff first (as it's already **_massive_**) and address this problem as part of https://github.com/getsentry/sentry-dotnet/pull/2452.

cc: @SeanFeldman